lib/SILGen/SILGenExpr.cpp

```diff
@@ -2380,7 +2380,35 @@ RValue RValueEmitter::visitAbstractClosureExpr(AbstractClosureExpr *e,
 RValue RValueEmitter::
 visitInterpolatedStringLiteralExpr(InterpolatedStringLiteralExpr *E,
                                    SGFContext C) {
-  return visit(E->getSemanticExpr(), C);
+  RValue result;
+  PreparedArguments interpolationInitArgs;
+  RValue literalCapacity = visit(E->getLiteralCapacityExpr(), SGFContext());
+  RValue interpolationCount = visit(E->getInterpolationCountExpr(), SGFContext());
```

Comment:
As a further cleanup, it might be nice to build integer literals right here in SILGen instead of attaching them to the AST. Probably not necessary for this PR, though. (And I don't know if we do this elsewhere for actual `Int`s, as opposed to builtin types; that could make it tricky.)

Reply:
I looked into doing it this way initially, but we currently do the protocol lookup in handleIntegerLiteralExpr which happens in lib/Sema. This suggestion would make it so that we have to move / duplicate some of that logic into SILGen. I suppose, technically, we could do the witness lookup in silgen as well for all exprs.

Reply:
This way is fine!